libmattermost.c

```diff
@@ -1071,111 +1105,222 @@ static void mm_socket_write_json(MattermostAccount *ma, JsonObject *data);
 static void mm_get_users_by_ids(MattermostAccount *ma, GList *ids);
 static void mm_get_avatar(MattermostAccount *ma, PurpleBuddy *buddy);
 
-static void mm_join_room(MattermostAccount *ma, const gchar *team_id, const gchar *channel_id);
+static void mm_join_room(MattermostAccount *ma, const gchar *team_id, MattermostChannel *channel);
+
+
+
+void 
+mm_g_free_mattermostchannel(gpointer ch)
+{
+	MattermostChannel *mc = ch;
+	g_free(mc->id);
+	g_free(mc->team_id);
+	g_free(mc->type);
+	g_free(mc->display_name);
+}
+
+int 
+mm_compare_channels_int(gconstpointer a, gconstpointer b)
+{
+	const MattermostChannel *p1 = a;
+	const MattermostChannel *p2 = b;
+
+	if (!g_strcmp0(p1->id,p2->id)) {
+		return 0;
+	}
+	return -1;
+}
+
+int
+mm_compare_channels_by_display_name_int(gconstpointer a, gconstpointer b)
+{
+	const MattermostChannel *p1 = a;
+	const MattermostChannel *p2 = b;
+
+	gint res = g_strcmp0(p1->display_name,p2->display_name);
+	
+	if (res < 0) { return 1;}
+	if (res > 0) { return -1;}
+
+	return 0;
+}
+
+int
+mm_compare_channels_by_name_int(gconstpointer a, gconstpointer b)
+{
+	const MattermostChannel *p1 = a;
+	const MattermostChannel *p2 = b;
+
+	gint res = g_strcmp0(p1->name,p2->name);
+	
+	if (res < 0) { return 1;}
+	if (res > 0) { return -1;}
+
+	return 0;
+}
+
+
+const gchar *
+mm_get_alias(MattermostUser *mu)
+{
+	gchar *nickname = NULL;
+	gchar *full_name = NULL;
+
+	if (mu->nickname && *mu->nickname) { nickname = g_strconcat(" (",mu->nickname,")",NULL); }
+	full_name = g_strconcat(mu->first_name ? mu->first_name : "", (mu->first_name && *mu->first_name) ? " " : "", mu->last_name, nickname,  NULL);
+    const gchar *alias = g_strdup((full_name && *full_name) ? full_name : (mu->email && *mu->email) ? mu->email : NULL);

```

Comment:
Don't use strdup for a const, leaks mem. Can't define alias half way through code unless c99 compile flag is set